nss-day-cohort-29/reactive-nutshell-copperden
# Chat Add-a-Friend Functionality

# Description
- Changed position of onClick for editing messages (from "edit" link to user's message itself).
- Added onClick to usernames (except the current user's) that enables the current user to add that user as a friend. (Adding friend doesn't actually POST - waiting on others to finish that - but the click is set up for it).
 
## Type of change

# Testing Instructions for Change Made
1. `git fetch --all`
2. `git checkout chat-edits`
3. Use the database data below.
4. Go to "Messages"
5. Click on _userName_'s purple bubble and check that edit functionality still works.
6 Click on _userName2_'s name and you should receive a dialog saying "You're already friends!"
7. Click on _notMyFriend_'s name and you should receive a prompt asking if you want to be friends.
8. Click Yes and you'll see it console log (to be replaced with actual POST).

```
{
  "users": [
    {
      "id": 1,
      "name": "userName",
      "email": "userEmail@gmail.com"
    },
    {
      "id": 2,
      "name": "userName2",
      "email": "userEmail2@gmail.com"
    },
    {
      "id": 3,
      "name": "notMyFriend",
      "email": "userEmail3@gmail.com"
    }
  ],
  "events": [],
  "articles": [],
  "tasks": [],
  "messages": [
    {
      "message": "User1 Message",
      "timeDisplay": "1/22/2019 at 3:18 PM",
      "timestamp": 1548710316504,
      "userId": 1,
      "id": 1
    },
    {
      "id": 2,
      "message": "User2 Message",
      "timeDisplay": "1/27/2019 at 3:18 PM",
      "timestamp": 1548710316504,
      "userId": 2
    },
    {
      "message": "User3 Message",
      "timeDisplay": "1/28/2019 at 8:28 PM",
      "timestamp": 1548728880201,
      "userId": 3,
      "id": 5
    }
  ],
  "friends": [
    {
      "id": 1,
      "currentUserId": 1,
      "userId": 2
    },
    {
      "id": 2,
      "currentUserId": 2,
      "userId": 1
    }
  ]
}
```

# Checklist:
